pennylane/ops/qubit/matrix_ops.py

```diff
@@ -389,6 +420,10 @@ def compute_matrix(D):  # pylint: disable=arguments-differ
         if not qml.math.allclose(D * qml.math.conj(D), qml.math.ones_like(D)):
             raise ValueError("Operator must be unitary.")
 
+        # The diagonal is supposed to have one-dimension. If it is broadcasted, it has two
+        if len(qml.math.shape(D)) == 2:
+            return qml.math.stack([qml.math.diag(_D) for _D in D])
```

Comment:
Confusing 😅

Reply:
Because of the condition in the `if` clause? :D Can I improve this anyhow?

Reply:
Oh, I think I was laughing about the two-dimensional diagonal. It makes sense, just took me by surprise!